lib/service_order_listener.rb

```diff
@@ -17,54 +17,57 @@ def run
   end
 
   def subscribe_to_task_updates
-    self.client = ManageIQ::Messaging::Client.open(messaging_client_options)
-
     Rails.logger.info("Catalog API service order listener started...")
 
-    client.subscribe_messages(
-      :service   => SERVICE_NAME,
-      :max_bytes => 500_000
-    ) do |messages|
-      messages.each do |msg|
-        process_message(msg)
+    ManageIQ::Messaging::Client.open(messaging_client_options) do |client|
+      client.subscribe_topic(
+        :service     => SERVICE_NAME,
+        :persist_ref => CLIENT_REF,
+        :max_bytes   => 500_000
+      ) do |topic|
+        process_event(topic)
       end
     end
-  ensure
-    Rails.logger.info("Catalog API service order listener stopping...")
-
-    client&.close
-    self.client = nil
   end
 
   private
 
-  def process_message(msg)
-    Rails.logger.info("Processing #{msg.message} with payload: #{msg.payload}")
+  def process_event(topic)
+    Rails.logger.info("Processing service order topic message: #{topic.message} with payload: #{topic.payload}")
 
     ProgressMessage.create!(
       :level   => "info",
-      :message => "Task update message received with payload: #{msg.payload}"
+      :message => "Task update message received with payload: #{topic.payload}"
```

Comment:
@mkanoor should  the `OrderItem`be looked up earlier so the `order_item_id` can be set on the `ProgressMessage` record?